tests/e2e/util/kubectl/kubectl.go

```diff
@@ -21,97 +21,91 @@ import (
+	"github.com/onsi/ginkgo/v2"
 )
 
+type KubectlBuilder struct {
+	binary     string
+	namespace  string
+	kubeconfig string
+}
+
 const DefaultBinary = "kubectl"
 
-// optionalKubeconfig add the flag --kubeconfig if the kubeconfig is set
-func optionalKubeconfig(kubeconfig []string) string {
-	if len(kubeconfig) > 0 && kubeconfig[0] != "" {
-		return fmt.Sprintf("--kubeconfig %s", kubeconfig[0])
-	}
-	return ""
+func newKubectlBuilder() *KubectlBuilder {
+	return &KubectlBuilder{}
 }
 
-// kubectl return the kubectl command
-// If the environment variable COMMAND is set, it will return the value of COMMAND
-// Otherwise, it will return the default value "kubectl" as default
-// Arguments:
-// - format: format of the command without kubeclt or oc
-// - args: arguments of the command
-func kubectl(format string, args ...interface{}) string {
+func (k *KubectlBuilder) setBinary() {
 	binary := DefaultBinary
 	if cmd := os.Getenv("COMMAND"); cmd != "" {
 		binary = cmd
 	}
 
-	return binary + " " + fmt.Sprintf(format, args...)
+	k.binary = binary
 }
 
-// CreateFromString creates a resource from the given yaml string
-func CreateFromString(yamlString string, kubeconfig ...string) error {
-	cmd := kubectl("create %s -f -", optionalKubeconfig(kubeconfig))
-	_, err := shell.ExecuteCommandWithInput(cmd, yamlString)
-	if err != nil {
-		return fmt.Errorf("error creating resource from yaml: %w", err)
-	}
-	return nil
+func (k *KubectlBuilder) setNamespace() {
+	k.namespace = ""
 }
 
-// ApplyString applies the given yaml string to the cluster
-func ApplyString(ns, yamlString string, kubeconfig ...string) error {
-	nsflag := nsflag(ns)
-	// If the namespace is empty, we need to remove the flag because it will fail
-	// TODO: improve the nsflag function to handle this case
-	if ns == "" {
-		nsflag = ""
+func (k *KubectlBuilder) setKubeconfig() {
+	k.kubeconfig = ""
+}
+
+func (k *KubectlBuilder) build(cmd string) string {
+	args := []string{k.binary}
+
+	// Only append namespace if it's set
+	if k.namespace != "" {
+		args = append(args, k.namespace)
 	}
 
-	cmd := kubectl("apply %s %s --server-side -f -", nsflag, optionalKubeconfig(kubeconfig))
-	_, err := shell.ExecuteCommandWithInput(cmd, yamlString)
-	if err != nil {
-		return fmt.Errorf("error applying yaml: %w", err)
+	// Only append kubeconfig if it's set
+	if k.kubeconfig != "" {
+		args = append(args, k.kubeconfig)
 	}
 
-	return nil
+	args = append(args, cmd)
+
+	// Join all the arguments with a space
+	ginkgo.GinkgoWriter.Write([]byte(fmt.Sprintf("Running command: %s\n", strings.Join(args, " "))))
+	return strings.Join(args, " ")
 }
 
-// Apply applies the given yaml file to the cluster
-func Apply(ns, yamlFile string, kubeconfig ...string) error {
-	err := ApplyWithLabels(ns, yamlFile, "", kubeconfig...)
-	return err
+// NewKubectlBuilder creates a new KubectlBuilder
+func NewKubectlBuilder() *KubectlBuilder {
+	k := newKubectlBuilder()
+	k.setBinary()
+	k.setKubeconfig()
```

Comment:
this doesn't do anything, does it?

Reply:
Updated by deleting `setNamespace()` and `setKubeconfig()`, now the NewKubectlBuilder func will return the `KubectlBuilder` with only set the Binary from the env var